datafusion/proto/tests/cases/roundtrip_logical_plan.rs

```diff
@@ -904,7 +904,7 @@ async fn roundtrip_expr_api() -> Result<()> {
             vec![lit(1), lit(2), lit(3)],
             vec![lit(10), lit(20), lit(30)],
         ),
-        row_number(vec![col("a")]),
```

Comment:
If we did the signature check, this test should fail.

The reason why it is not is because we didn't do signature check for window udf.

We have similar functions for scalar and udaf.

https://github.com/apache/datafusion/blob/cb1e3f0e5cfd04195ae0c933b74f3eb3fcbe1b45/datafusion/expr/src/type_coercion/functions.rs#L42-L96

`data_types_with_scalar_udf` and `data_types_with_aggregate_udf`.

After adding a new `data_types_with_window_udf` function, we should get the failed msg for incorrect non-empty args for `row_number`

Reply:
I believe this was done in https://github.com/apache/datafusion/pull/12045